verification/verifier_helpers.go

```diff
@@ -0,0 +1,53 @@
+package verification
+
+import (
+	"crypto/x509"
+	"fmt"
+	"strings"
+)
+
+// isFailureResult checks whether a VerificationResult fails the entire signature verification workflow.
+// signature verification workflow is considered failed if there is a VerificationResult with "Enforced" as the action but the result was inconclusive (failed to verify) or unsuccessful
+func isFailureResult(result VerificationResult) bool {
+	return result.Action == Enforced && (result.FailedToVerify || !result.Success)
+}
+
+func verifyX509TrustedIdentities(certs []*x509.Certificate, trustPolicy *TrustPolicy) error {
+	if isPresent(wildcard, trustPolicy.TrustedIdentities) {
+		return nil
+	}
+
+	var trustedX509Identities []map[string]string
```

Comment:
IMO it's better to have a strongly types DN struct which contains the RDN map, than using nested map/array of more than two level.